main.go

```diff
@@ -56,11 +56,11 @@ func initProviderCreator() provider.ProviderCreator {
 		)
 	}
 
-	if !viper.IsSet("user") {
+	if !viper.IsSet("user") && !viper.IsSet("authorization-lambda") {
 		log.Print("Warning: No user specified. \"user\" will be used")
 	}
 	viper.SetDefault("user", "user")
-	if !viper.IsSet("password") {
+	if !viper.IsSet("password") && !viper.IsSet("authorization-lambda") {
 		log.Print("Warning: No password specified. \"changeit\" will be used. DO NOT USE IN PRODUCTION!")
 	}
 	viper.SetDefault("password", "changeit")
```

Comment:
if you move these lines below the `if viper.IsSet("authorization-lambda") { ... }` you would not need to check if `viper.IsSet("authorization-lambda")` to suppress the warning

Reply:
I moved the code up. Good suggestion. Thanks. Rather than trying to piggyback on your staticCreds.S3Bucket, I just create a local S3Bucket var and pass that in to my ProviderCreator. Do you see anything wrong with that approach? Personally, I think it is a little cleaner than what I was trying to do.